java/datastore/src/main/java/com/google/datastore/v1/client/RemoteRpc.java

```diff
@@ -249,6 +249,12 @@ static DatastoreException makeException(String url, String methodName, InputStre
           String.format(
               "Unable to parse Status protocol buffer: HTTP status code was %s.", httpStatusCode),
           e);
+    } finally {
+      try {
+        content.close();
+      } catch(IOException e) {
+        logger.fine(String.format("closing content stream failed: %s",e));
+      }
```

Comment:
I think we can also fail to close the stream in the try block that starts at line 231. What about instead adding a try-finally around the code at line 185. That should ensure the stream is closed no matter what happens in this method.

Reply:
I updated the pull request based on your suggestion, and also modified the `IOUtils.copy` line to avoid a potential double-close.